sklearn/linear_model/logistic.py

```diff
@@ -1198,6 +1202,14 @@ def fit(self, X, y, sample_weight=None):
         self : object
             Returns self.
         """
+        if self.solver == 'auto' and self.penalty == 'l2':
+            self.solver = 'liblinear'
```

Comment:
We don't allow fit to change constructor parameters. You can store the validated parameter as _solver if it should be private, or solver_ if it should be public (assumed useful to users).